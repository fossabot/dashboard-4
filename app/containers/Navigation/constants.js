/*
 *
 * Navigation constants
 *
 */

export const DRAWER_OPEN = 'efr-frontend/Navigation/DRAWER_OPEN';
export const DRAWER_CLOSE = 'efr-frontend/Navigation/DRAWER_CLOSE';
