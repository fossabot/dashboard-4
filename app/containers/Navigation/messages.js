/*
 *
 * Navigation messages
 *
 */

import { defineMessages } from 'react-intl';

export const scope = 'efr-frontend.containers.Navigation';

export default defineMessages({
  appbarHeader: {
    id: `${scope}.appbar.header`,
    defaultMessage: 'Enterprise Flow Repository',
  },
});
