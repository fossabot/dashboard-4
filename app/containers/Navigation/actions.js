/*
 *
 * Navigation actions
 *
 */

import { DRAWER_OPEN, DRAWER_CLOSE } from './constants';

/**
 * Open the header drawer.
 *
 * @return {object} An action object with a type of HEADER_OPEN
 */
export function drawerOpen() {
  return {
    type: DRAWER_OPEN,
  };
}

/**
 * Open the header drawer.
 *
 * @return {object} An action object with a type of HEADER_OPEN
 */
export function drawerClose() {
  return {
    type: DRAWER_CLOSE,
  };
}
