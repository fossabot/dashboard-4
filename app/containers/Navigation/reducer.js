/*
 *
 * Navigation reducer
 *
 */

import produce from 'immer';
import { DRAWER_OPEN, DRAWER_CLOSE } from './constants';

// The initial state of the App
export const initialState = {
  drawer: {
    open: true,
  },
};

/* eslint-disable default-case, no-param-reassign */
const navigationReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case DRAWER_OPEN:
        draft.drawer.open = true;
        break;

      case DRAWER_CLOSE:
        draft.drawer.open = false;
        break;
    }
  });

export default navigationReducer;
