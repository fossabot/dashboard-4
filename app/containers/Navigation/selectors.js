/*
 *
 * Navigation selectors
 *
 */

import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the test state domain
 */

const selectNavigationDomain = state => state.navigation || initialState;

/**
 * Other specific selectors
 */

const makeSelectDrawerOpen = () =>
  createSelector(
    selectNavigationDomain,
    navigationState => navigationState.drawer.open,
  );

export default selectNavigationDomain;
export { makeSelectDrawerOpen };
