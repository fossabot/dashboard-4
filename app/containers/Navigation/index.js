/*
 *
 * Navigation
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage } from 'react-intl';

import clsx from 'clsx';
import { useTheme } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import InboxIcon from '@material-ui/icons/MoveToInbox';
import MailIcon from '@material-ui/icons/Mail';

import messages from './messages';
import { makeSelectDrawerOpen } from './selectors';
import { drawerOpen, drawerClose } from './actions';

import MaterialStyles from '../../material-styles';

export function Navigation({
  drawerOpenSelector,
  onDrawerOpen,
  onDrawerClose,
}) {
  const materialClasses = MaterialStyles();
  const materialTheme = useTheme();

  return (
    <div>
      <AppBar
        position="fixed"
        className={clsx(materialClasses.appBar, {
          [materialClasses.appBarShift]: drawerOpenSelector,
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={onDrawerOpen}
            edge="start"
            className={clsx(materialClasses.menuButton, {
              [materialClasses.hide]: drawerOpenSelector,
            })}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" noWrap>
            <FormattedMessage {...messages.appbarHeader} />
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer
        variant="permanent"
        className={clsx(materialClasses.drawer, {
          [materialClasses.drawerOpen]: drawerOpenSelector,
          [materialClasses.drawerClose]: !drawerOpenSelector,
        })}
        classes={{
          paper: clsx({
            [materialClasses.drawerOpen]: drawerOpenSelector,
            [materialClasses.drawerClose]: !drawerOpenSelector,
          }),
        }}
        open={drawerOpenSelector}
      >
        <div className={materialClasses.toolbar}>
          <IconButton onClick={onDrawerClose}>
            {materialTheme.direction === 'rtl' ? (
              <ChevronRightIcon />
            ) : (
              <ChevronLeftIcon />
            )}
          </IconButton>
        </div>
        <Divider />
        <List>
          {['Inbox', 'Starred', 'Send email', 'Drafts'].map((text, index) => (
            <ListItem button key={text}>
              <ListItemIcon>
                {index % 2 === 0 ? <InboxIcon /> : <MailIcon />}
              </ListItemIcon>
              <ListItemText primary={text} />
            </ListItem>
          ))}
        </List>
        <Divider />
        <List>
          {['All mail', 'Trash', 'Spam'].map((text, index) => (
            <ListItem button key={text}>
              <ListItemIcon>
                {index % 2 === 0 ? <InboxIcon /> : <MailIcon />}
              </ListItemIcon>
              <ListItemText primary={text} />
            </ListItem>
          ))}
        </List>
      </Drawer>
    </div>
  );
}

Navigation.propTypes = {
  drawerOpenSelector: PropTypes.bool,
  onDrawerOpen: PropTypes.func,
  onDrawerClose: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  drawerOpenSelector: makeSelectDrawerOpen(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onDrawerOpen: () => dispatch(drawerOpen()),
    onDrawerClose: () => dispatch(drawerClose()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  memo,
)(Navigation);
