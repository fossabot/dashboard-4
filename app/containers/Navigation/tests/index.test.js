/**
 *
 * Tests for Navigation
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Navigation, { mapDispatchToProps } from '../index';
import { drawerOpen, drawerClose } from '../actions';
import configureStore from '../../../configureStore';

import { DEFAULT_LOCALE } from '../../../i18n';

describe('<Navigation />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Navigation dispatch={dispatch} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Navigation />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onDrawerOpen', () => {
      it('Should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onDrawerOpen).toBeDefined();
      });

      it('Should dispatch drawerOpen when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onDrawerOpen();
        expect(dispatch).toHaveBeenCalledWith(drawerOpen());
      });
    });

    describe('onDrawerClose', () => {
      it('Should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onDrawerClose).toBeDefined();
      });

      it('Should dispatch drawerClose when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onDrawerClose();
        expect(dispatch).toHaveBeenCalledWith(drawerClose());
      });
    });
  });
});
