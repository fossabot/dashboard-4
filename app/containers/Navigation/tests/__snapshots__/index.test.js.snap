// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Should render and match the snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-162 makeStyles-appBarShift-163 MuiAppBar-colorPrimary mui-fixed"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-164 makeStyles-hide-165 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
      >
        <span>
          Enterprise Flow Repository
        </span>
      </h6>
    </div>
  </header>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-166 makeStyles-drawerOpen-167"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper makeStyles-drawerOpen-167 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
    >
      <div
        class="makeStyles-toolbar-169"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <ul
        class="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Inbox
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Starred
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Send email
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Drafts
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
      <hr
        class="MuiDivider-root"
      />
      <ul
        class="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              All mail
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Trash
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Spam
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
  </div>
</div>
`;
